### PR TITLE
fix: use external-secret standard naming

### PR DESCRIPTION
**Pull request by dali546**
#### Description
renamed parameter-store to secret-manager as ExternalSecrets uses that
convention.

#### Special notes for the reviewer(s)
related: https://github.com/jenkins-x/jx3-versions/pull/2619

### CODE REVIEW

File: modules/cluster/irsa.tf [hunk 1]
```diff
@@ -421,7 +421,7 @@ module "iam_assumable_role_secrets-secrets-manager" {
 // ----------------------------------------------------------------------------
 // External Secrets - Parameter Store
 // ----------------------------------------------------------------------------
-data "aws_iam_policy_document" "parameter-store-policy" {
+data "aws_iam_policy_document" "system-manager-policy" {
```
Comment (ankitm123) [inline modules/cluster/irsa.tf:424]: Why this change? since the other one is called secrets manager, and outside system manager do we have to make this change?

Comment (dali546) [reply modules/cluster/irsa.tf:424]: so this change is purely to be uniform with the rest of external secrets. external-secrets calls it "systemManager", so just so it's the same everywhere, I want to change it here too
